### PR TITLE
Separate qca-swiss-army-knife from firmware-qcom-dragonboard845c

### PR DESCRIPTION
**Pull request by lumag**
On RB5 ath11k module requires board data not present in linux-firmware archive. Add support for generating it at build time.

### CODE REVIEW

**Comment by ndechesne** [comment]
run tests

**Comment by ndechesne** [comment]
looks good to me. @alimon can you check as well?

**Comment by lumag** [comment]
run tests

**Comment by alimon** [comment]
> looks good to me. @alimon can you check as well?

It looks fine to me, I will build this PR to review that loads properly in DB845c.

**Comment by lumag** [comment]
Updated patches to follow linux-firmware-ath11k appearing in oe-core layer

File: recipes-bsp/firmware/firmware-qcom-rb5_1.0.bb [hunk 1]
```diff
@@ -53,6 +71,9 @@ do_install() {
         install -m 0444 slpi.b* slpi.mdt ${D}${nonarch_base_libdir}/firmware/qcom/sm8250
         install -m 0444 venus.b* venus.mdt ${D}${nonarch_base_libdir}/firmware/qcom/sm8250
 
+        install -d ${D}${nonarch_base_libdir}/firmware/ath11k/QCA6390/hw2.0/
+        install -m 0444 ${S}/board-2.bin ${D}${nonarch_base_libdir}/firmware/ath11k/QCA6390/hw2.0/board-2.bin
```
Comment (ndechesne) [inline recipes-bsp/firmware/firmware-qcom-rb5_1.0.bb:75]: what happens if NHLOS_URI is not provided? would board-2.bin be missing?

Comment (ndechesne) [reply recipes-bsp/firmware/firmware-qcom-rb5_1.0.bb:75]: oops. sorry. ignore.

**Comment by kraj** [comment]
this is causing some packaging gripes see

https://errors.yoctoproject.org/Errors/Details/539389/
https://errors.yoctoproject.org/Errors/Details/539387/

**Pull request merged**